src/main/java/com/osiris/autoplug/client/tasks/updater/server/TaskServerUpdater.java

```diff
@@ -403,114 +300,30 @@ private void doFabricUpdatingLogic() throws WrongJsonTypeException, IOException,
-                        if (download.compareWithSHA256(build_hash)) {
-                            File final_dest = serverExe;
-                            if (final_dest == null)
-                                final_dest = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
-                            if (final_dest.exists()) final_dest.delete();
-                            final_dest.createNewFile();
-                            FileUtils.copyFile(cache_dest, final_dest);
-                            setStatus("Server update was installed successfully (" + build_id + " -> " + latest_build_id + ")!");
-                            updaterConfig.server_build_id.setValues("" + latest_build_id);
-                            updaterConfig.save();
-                            setSuccess(true);
-                        } else {
-                            if (updaterConfig.server_skip_hash_check.asBoolean()) {
-                                addWarning("Note that the hash check failed for this download, but installing anyways because skip-hash-check is enabled!");
-                                File final_dest = serverExe;
-                                if (final_dest == null)
-                                    final_dest = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
-                                if (final_dest.exists()) final_dest.delete();
-                                final_dest.createNewFile();
-                                FileUtils.copyFile(cache_dest, final_dest);
-                                setStatus("Server update was installed successfully (" + build_id + " -> " + latest_build_id + ")!");
-                                updaterConfig.server_build_id.setValues("" + latest_build_id);
-                                updaterConfig.save();
-                                setSuccess(true);
-                            } else {
-                                setStatus("Downloaded server update is broken (hash check failed). Nothing changed!");
-                                setSuccess(false);
-                            }
-                        }
-
-                    } else {
-                        setStatus("Server update failed!");
-                        setSuccess(false);
-                    }
-                    break;
-                }
+            setStatus("Starting server updating...");
+            File outputFile;
+            if (profile.equals("MANUAL")) {
+                outputFile = new File(downloadsDir.getAbsolutePath() + "/" + serverSoftware + "-latest.jar");
+            } else if (serverExe == null) {
+                outputFile = new File(GD.WORKING_DIR + "/" + serverSoftware + "-latest.jar");
+            } else {
+                outputFile = serverExe;
+            }
+            try {
+                UpdateStatus status = UpdateBuilder.updateProject(serverSoftware)
+                        .version(serverVersion)
```

Comment:
The server version is the Minecraft version, how does it update between builds for the same version?

Reply:
> The server version is the Minecraft version, how does it update between builds for the same version?

It's MCServerUpdater dude. It has `getLatestBuild`.